contracts/TicTacToe.sol

```diff
@@ -8,21 +8,22 @@ contract TicTacToe {
 		address player2;
 	}
 
-	Game[] public games;
+	uint public gameCounter;
+	mapping (uint => Game) public games;
 
 	function newGame(address player2) public returns(uint) {
 		// Creates a new game with player1 set to msg.sender
-		
-		address player1 = msg.sender;
+		gameCounter++;
 		uint8[9] memory newBoard;
 
-		Game memory game = Game(newBoard, player1, player2);
-		uint id = games.push(game) - 1;
```

Comment:
Do you know why this didn't work? What I saw in my tests was that this always returned `id = 0`. It feels like since `games` is `storage`, if I keep calling `newGame`, the length of `games` should increase and so should the returned `id`, but when I tried this in `truffle develop`, it isn't what I saw...

Reply:
Not completely sure as I did not have time to dig deeper.  But I noticed the behavior of returning only zero. Perhaps it has to do that we are trying to store a var from memory `game` to the array storage `Game[]`?

Also I did not assume much on the tic tac toe game history specification. Is it a one-off game that we don't care about the previous games or game history is important to show user their past games?  If you want to store all the user's games,  a mapping from address to an array of Game structs could work  i.e. `mapping (address => Game[]) public games`